src/main/preload.js

```diff
@@ -15,6 +15,15 @@ require('angular-sanitize');
 require('angular-local-storage');
 require('angular-qrcode');
 require('angular-zxcvbn');
+window.react = require('react');
```

Comment:
Why are we attaching these to `window` instead of `require`ing them where necessary?

Reply:
@dfcowell that is because of the `nodeIntegration` false. It won't require the dependency in run time. I mean on the renderer process. So I need to require it on main process through the preload. That sucks I know but that I think it the way to go if we want more security.

Reply:
Why does a simple `require` need `nodeIntegration` if we're bundling it with Webpack? This is what webpack is designed for.

Reply:
@dfcowell yeah, it works differently with electron. With  nodeIntegration false it want allow any code from the renderer process to make a require to node modules. Only to modules on with code inside the renderer process. Otherwise anyone could require through the dev tools console or inject a require code somewhere. So it protects from that. Usually electron apps don't need to set nodeIntegration false so it is really hard to find material on the internet on how to handle node modules on renderer process when it is false. I tried many different ways to get the modules loaded but that was the only way that it worked. Cheers

Reply:
There is no way that the only way to load modules in the renderer in Electron is to bind them all to the window object.

It's a web browser. If we run webpack to build an app bundle there's no reason it wouldn't work. I'll fix this one myself.

Reply:
@rodrigopavezi I've done a lot more research. We have two options:

1. *Set `nodeIntegration` to `true`:* There's no reason we need it set to `false` for the main window of the application. All that runs in that window is local, trusted code. I prefer this option.
2. *Extend `electron-webpack` as a library and override its `target` for the renderer build, making it treat the renderer like a standard web SPA - which would mean bundling its own `require` and all of the various modules. This adds a lot of maintenance overhead, but we would be able to very strictly control what the renderer has access to.

I think we should only go with option 2 if there is a compelling reason, which I personally can't find.

Reply:
@dfcowell this guide has the reason why we have chosen to set nodeIntegration to false.

Reply:
That specifically says "for untrusted origins."

It's specifically talking about loading http://some-remote-untrusted-webpage.com/whatever in an electron context.

We don't do that.

Reply:
@dfcowell good point, so we are just over thinking here and being too protective in a way that hurt us :-) funny. Okay I will set it to true and remove all this preload require stuff. Do you agree? Cheers